search/searcher/conjunction.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package searcher
+
+import (
+	"github.com/m3db/m3ninx/postings"
+	"github.com/m3db/m3ninx/search"
+)
+
+type conjunctionSearcher struct {
+	searchers search.Searchers
+	len       int
+
+	idx  int
+	curr postings.List
+
+	closed bool
+	err    error
+}
+
+// NewConjunctionSearcher returns a new Searcher which matches documents which match each
+// of the given Searchers. It is not safe for concurrent access.
+func NewConjunctionSearcher(ss search.Searchers) (search.Searcher, error) {
```

Comment:
same question as for the disjunction. Why do it piecemeal instead of all at once